src/core/properties/listproperty.cpp

```diff
@@ -0,0 +1,240 @@
+    std::vector<std::unique_ptr<Property>> result;
+    result.reserve(v.size());
+    std::transform(v.begin(), v.end(), std::back_inserter(result),
+                   [](auto& p) { return std::unique_ptr<Property>(p->clone()); });
+
+    return result;
+}
+
+}  // namespace detail
+
+ListProperty::ListProperty(std::string identifier, const std::string& displayName,
+                           size_t maxNumberOfElements, ListPropertyUIFlags uiFlags,
+                           InvalidationLevel invalidationLevel, PropertySemantics semantics)
+    : CompositeProperty(identifier, displayName, invalidationLevel, semantics)
+    , uiFlags_(uiFlags)
+    , maxNumElements_("maxNumberOfElements", maxNumberOfElements) {}
+
+ListProperty::ListProperty(std::string identifier, const std::string& displayName,
+                           std::vector<std::unique_ptr<Property>>&& prefabs,
+                           size_t maxNumberOfElements, ListPropertyUIFlags uiFlags,
+                           InvalidationLevel invalidationLevel, PropertySemantics semantics)
+    : ListProperty(identifier, displayName, maxNumberOfElements, uiFlags, invalidationLevel,
+                   semantics) {
+    prefabs_ = std::move(prefabs);
+}
+
+ListProperty::ListProperty(std::string identifier, const std::string& displayName,
+                           std::unique_ptr<Property>&& prefab, size_t maxNumberOfElements,
+                           ListPropertyUIFlags uiFlags, InvalidationLevel invalidationLevel,
+                           PropertySemantics semantics)
+    : ListProperty(identifier, displayName, maxNumberOfElements, uiFlags, invalidationLevel,
+                   semantics) {
+    prefabs_.emplace_back(std::move(prefab));
+}
+
+ListProperty::ListProperty(const ListProperty& rhs)
+    : CompositeProperty(rhs)
+    , uiFlags_(rhs.uiFlags_)
+    , maxNumElements_(rhs.maxNumElements_)
+    , prefabs_(std::move(detail::clonePropertyVector(rhs.prefabs_))) {}
+
+ListProperty& ListProperty::operator=(const ListProperty& that) {
+    if (this != &that) {
+        uiFlags_ = that.uiFlags_;
+        maxNumElements_ = that.maxNumElements_;
+        prefabs_ = std::move(detail::clonePropertyVector(that.prefabs_));
+    }
+    return *this;
+}
+
+ListProperty* ListProperty::clone() const { return new ListProperty(*this); }
+
+std::string ListProperty::getClassIdentifierForWidget() const {
+    return ListProperty::CLASS_IDENTIFIER;
+}
+
+void ListProperty::set(const Property* src) {
+    if (const auto listproperty = dynamic_cast<const ListProperty*>(src)) {
+        set(listproperty);
+    }
+}
+
+void ListProperty::set(const ListProperty* src) {
+    NetworkLock l(this);
+    // check for matching prefab types first
+    if (getPrefabIDs() == src->getPrefabIDs()) {
+        // TODO: should we sync/consider the UI flags here as well?
+
+        maxNumElements_ = src->maxNumElements_;
+
+        while (size()) {
+            removeProperty(static_cast<size_t>(0));
```

Comment:
What are you doing here?

Reply:
Only need to clone it the types are different, the default should be jpto just call set on the sub properties

Reply:
This was the first draft for linking. Open for suggestions.
But when cloning only non-matching properties, PropertyOwner needs functions for inserting properties, i.e. `insertProperty()`, so the order is similar.

Reply:
You can fall back on cloning all if one is different... linking appens alot and should be as cheap as possible...

Reply:
Cloning also has severe downsides. I tried it with TF properties in the list. It horribly failed, due to missing callbacks, wrong VolumePort, etc.

Reply:
I think callbacks are a general problem when "cloning" properties. @r-englund mentioned something similar.

Reply:
List elements are now matched based on their class identifier. Missing elements are cloned while superfluous list entries are deleted.